src/Illuminate/Database/Query/Builder.php

```diff
@@ -682,7 +682,7 @@ public function where($column, $operator = null, $value = null, $boolean = 'and'
         // If the column is making a JSON reference we'll check to see if the value
         // is a boolean. If it is, we'll add the raw boolean string as an actual
         // value to the query to ensure this is properly handled by the query.
-        if (Str::contains($column, '->') && is_bool($value)) {
+        if (Str::contains((string) $column, '->') && is_bool($value)) {
```

Comment:
Why do we need this when `Str::contains()` already check for `is_string()` and return false.

Reply:
As I explained in this line, one test case passed a `Illuminate\Database\Query\Expression` as a column, which would evaluate to `false` if it wasn't cast to a string.

The test case is the following:

https://github.com/laravel/framework/blob/49f9db00a754517f07d365ee5b0806df82e37029/tests/Database/DatabaseQueryBuilderTest.php#L2612

If not cast to a string, `$column = new Raw("items->'$.available'")`, and when calling `Str::contains($column, '->')` would evaluate to false.

Reply:
Hope it is clear now. Maybe it wasn't clear enough on my PR description, my English is not the best, but I am working on improving it =)